egs/uw3/v1/local/prepare_lm.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# Copyright 2015     Hossein Hadian
+# Apache 2.0
+
+
+ngram=2
+
+[ -f ./path.sh ] && . ./path.sh  # source the path.
+. parse_options.sh || exit 1
+
+# This is only called for decoding purposes. This is not necessary for training.
+if [ $# != 2 ]; then
+  echo "Usage: $0 [--ngram <n-gram-order>] <lm-src> <out-lang-dir>"
+  echo "<lm-src> can either be an ARPA file or transcriptions file, e.g. train/text"
+  exit 1;
+fi
+
+lmsrc=$1
+lang=$2
+
+tmpdir=$lang/temp
+mkdir -p $tmpdir
+
+if [[ $lmsrc == *.arpa ]]; then
+  echo "$0: Using arpa LM: "$lmsrc
+  arpa=$lmsrc
+else
+  echo "$0: No Arpa file provided..Creating a $ngram-gram LM from provided text: "$lmsrc;
+  if [ -z $IRSTLM ]; then
+    export IRSTLM=$PWD/../../../tools/irstlm
+    export PATH=$PWD/../../../tools/irstlm/bin:$PATH
+    echo "$0: Assuming irstlm is at: "$IRSTLM
+  fi
+  ! which build-lm.sh 2>/dev/null && \
+    echo "$0: IRSTLM does not seem to be installed (build-lm.sh not on your path): " && \
+    echo "go to <kaldi-root>/tools and try 'make irstlm_tgt'" && exit 1;
+
+  cut -d' ' -f2- $lmsrc | sed -e 's:^:<s> :' -e 's:$: </s>:' \
+                              > $tmpdir/tmp_lm_train
+  build-lm.sh -k 1 -i $tmpdir/tmp_lm_train -n $ngram -o $tmpdir/tmp_lm.ilm.gz
+
+  compile-lm $tmpdir/tmp_lm.ilm.gz -t=yes /dev/stdout | \
+    grep -v unk > $tmpdir/lm_phone_bg.arpa
+
+  arpa=$tmpdir/lm_phone_bg.arpa
+fi
+
+cat $arpa | utils/find_arpa_oovs.pl $lang/words.txt  > $tmpdir/oovs.txt
+
+cat $arpa | \
```

Comment:
I think more preferable way would be just call utils\format_lm.sh (I noticed in the previous recipe)

Reply:
Also, it seems you are osciallating between IRSTLM and pocolm? is that necessary? Cannot you use just one toolkit?

Reply:
Yes, this is very old. I'll fix it.
Re LM toolkits, I'm not sure which toolkit to use.
Here (i.w. UW3) we just need a simple LM trained on training text.
But in IAM, we have 3 LM sources and pocolm might be more suited.

Reply:
I think I'm OK with using 2 different LM toolkits since it's 2 different recipes.  The only benefit in having a standard is if it's globally enforced, and that isn't practical for various reasons (there just isn't a clear leader).